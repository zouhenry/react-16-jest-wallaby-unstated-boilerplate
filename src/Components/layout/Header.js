import React from 'react';
import classNames from 'classnames';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import LeftNavContainer from "../../Containers/LeftNavContainer";

import MenuIcon from '@material-ui/icons/Menu';
import {Subscribe} from "unstated";


export default function Header({classes}) {
    return (
        <Subscribe to={[LeftNavContainer]}>
            {LeftNav => (
                <AppBar
                    position="fixed"
                    className={classNames(classes.appBar, {
                        [classes.appBarShift]: LeftNav.state.open,
                    })}
                >
                    <Toolbar disableGutters={!LeftNav.state.open}>
                        <IconButton
                            color="inherit"
                            aria-label="Open drawer"
                            onClick={LeftNav.handleDrawerOpen}
                            className={classNames(classes.menuButton, {
                                [classes.hide]: LeftNav.state.open,
                            })}
                        >
                            <MenuIcon/>
                        </IconButton>
                        <Typography variant="h6" color="inherit" noWrap>
                            My App
                        </Typography>
                    </Toolbar>
                </AppBar>
            )}
        </Subscribe>
    );
}
