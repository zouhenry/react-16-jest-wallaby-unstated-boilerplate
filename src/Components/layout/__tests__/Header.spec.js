import React from 'react';
import Header from '../Header';
import {render,mount, shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Provider} from "unstated";

configure({ adapter: new Adapter() });
describe('Header', function () {
    it('should match snapshot', ()=>{
        const actual = render(<Provider><Header classes={{}}></Header></Provider>);
        expect(toJson(actual)).toMatchSnapshot();
    })
});
