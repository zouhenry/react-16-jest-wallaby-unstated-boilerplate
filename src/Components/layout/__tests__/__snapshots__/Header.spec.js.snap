// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header
  class="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionFixed-2 MuiAppBar-colorPrimary-8 mui-fixed"
>
  <div
    class="MuiToolbar-root-37 MuiToolbar-regular-39"
  >
    <button
      aria-label="Open drawer"
      class="MuiButtonBase-root-47 MuiIconButton-root-41 MuiIconButton-colorInherit-42"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label-46"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-50"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
    </button>
    <h6
      class="MuiTypography-root-59 MuiTypography-h6-76 MuiTypography-colorInherit-88 MuiTypography-noWrap-85"
    >
      My App
    </h6>
  </div>
</header>
`;
